pkg/sentry/socket/epsocket/stack.go

```diff
@@ -138,3 +140,24 @@ func (s *Stack) TCPSACKEnabled() (bool, error) {
 func (s *Stack) SetTCPSACKEnabled(enabled bool) error {
 	return syserr.TranslateNetstackError(s.Stack.SetTransportProtocolOption(tcp.ProtocolNumber, tcp.SACKEnabled(enabled))).ToError()
 }
+
+// Forwarding implements inet.Stack.Forwarding.
+func (s *Stack) Forwarding(protocol tcpip.NetworkProtocolNumber) bool {
+	switch protocol {
+	case ipv4.ProtocolNumber, ipv6.ProtocolNumber:
+		return s.Stack.Forwarding(protocol)
+	default:
+		panic(fmt.Sprintf("unsupported protocol %v", protocol))
+	}
+}
+
+// SetForwarding implements inet.Stack.SetForwarding.
+func (s *Stack) SetForwarding(protocol tcpip.NetworkProtocolNumber, enable bool) error {
+	switch protocol {
+	case ipv4.ProtocolNumber, ipv6.ProtocolNumber:
+		s.Stack.SetForwarding(protocol, enable)
+	default:
+		panic(fmt.Sprintf("unsupported protocol %v", protocol))
```

Comment:
Should this not return an error?

Reply:
The motivation behind panic here is the following: a user can't pass the protocol variable, and it's up to a developer to pick up a protocol number. For example, the developer executes SetForwarding(arp.ARPProtocolNumber), and arp protocol doesn't support forwarding. Does it make sense to handle a returned error? Panic will help the developer to understand that her intention is wrong quickly.

Reply:
You're already returning an error from this method so it needs to be handled, and it looks like you're panicking in the caller anyway?

Reply:
fixed